testsuite/tests/basic/eval_order_pr10283.ml

```diff
@@ -0,0 +1,12 @@
+(* TEST *)
+
+(* Slightly modified version of an example from github user @Ngoguey42,
+   submitted as issue number 10283. *)
+
+let[@inline never][@local never] g () =
+  let f a b = Printf.printf "%d %d\n" a b in
```

Comment:
Does it make sense to tag this function `[@local]` to emphasize that interesting things happen because it gets simplified away?